### PR TITLE
Update the charm release process documentation.

### PR DESCRIPTION
**Pull request by frankban**
Remove unrequired steps.

### CODE REVIEW

**Comment by jujugui** [comment]
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/2699/
Test FAILed.

File: docs/process.rst [hunk 1]
```diff
@@ -281,8 +281,8 @@ Utopic) will run into problems with running the functional tests.
     ``bzr merge ../trusty-release``.
   - If required, commit the changes by running the following:
     ``bzr ci -m "Merged changes from the released charm."``.
-  - Copy the new GUI release to the releases directory of the charm
-    (i.e. ``develop-trunk/releases``).
+  - If a new GUI tarball is actually available, copy the new GUI release to the
```
Comment (kadams54) [inline docs/process.rst]: For my own curiosity, when would a new GUI tarball not be available?

Comment (frankban) [inline docs/process.rst]: When we are following those instructions to release the charm itself, not the GUI and the corresponding new charm.

File: docs/process.rst [hunk 2]
```diff
@@ -343,7 +339,7 @@ Functional test the GUI in two EC2 environments:
     suite, fix the errors before proceeding. If it ends up not being a trivial
     fix, stop this process and create a critical bug/card.
   - If everything went well, publish the new precise and trusty releases by
-    pushing the branch to the respective locations:
+    pushing the ``trusty-release`` branch to the respective locations:
```
Comment (kadams54) [inline docs/process.rst]: For some reason this section of bullet points is rendering as a blockquote - could that be fixed as a drive-by?

**Comment by kadams54** [comment]
:+1: with the comments above.

**Comment by jujugui** [comment]
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/2700/
Test FAILed.

File: docs/process.rst (hunk 2 above)
Comment (frankban) [inline docs/process.rst]: Done.

**Comment by jujugui** [comment]
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/2701/
Test FAILed.

**Comment by frankban** [comment]
Trying to ship this, thanks for the reviews.
:shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci.jujugui.org:8080/job/juju-gui-merge

**Comment by jujugui** [comment]
Build failed: Attempt to land pull request failed
        build url: http://ci.jujugui.org:8080/job/juju-gui-merge/947

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci.jujugui.org:8080/job/juju-gui-merge

**Comment by jujugui** [comment]
Build failed: Attempt to land pull request failed
        build url: http://ci.jujugui.org:8080/job/juju-gui-merge/948

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci.jujugui.org:8080/job/juju-gui-merge

**Comment by jujugui** [comment]
Build failed: Attempt to land pull request failed
        build url: http://ci.jujugui.org:8080/job/juju-gui-merge/949

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci.jujugui.org:8080/job/juju-gui-merge

**Comment by jujugui** [comment]
Build failed: Attempt to land pull request failed
        build url: http://ci.jujugui.org:8080/job/juju-gui-merge/950

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci.jujugui.org:8080/job/juju-gui-merge

**Comment by jujugui** [comment]
Build failed: Attempt to land pull request failed
        build url: http://ci.jujugui.org:8080/job/juju-gui-merge/951

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci.jujugui.org:8080/job/juju-gui-merge
